src/Appointments/Appointments.js

```diff
@@ -7,13 +7,45 @@ import "./Appointments.scss";
 
 const Appointments = ({ setAuthReg }) => {
   const [data, setData] = useState([]);
-  const [length, setLength] = useState(data.length);
   const [characters, setCharacters] = useState([]);
   const [flag, setFlag] = useState(true);
   const [isChange, setChange] = useState(false);
-  const [allRows, setAllRows] = useState(0);
   const [currentPage, setCurrentPage] = useState(1);
   const [rowsOnPage, setRowsOnPage] = useState(5);
+  const [allRows, setAllRows] = useState(0);
+  const [date, setDate] = useState({
+    before: "0000-00-00",
+    after: "9999-99-99",
+  });
+  const [sortItem, setSortItem] = useState({
+    value: "",
+    sort: "asc",
+  });
+
+  const reFlag = () => setFlag(!flag);
+
+  const reChange = () => setChange(!isChange);
+
+  const props = {
```

Comment:
it's considered improperly thought-out logic, if you need so many props
please think about it